### PR TITLE
delete from s.nodeByPayload as well when pruneFinalizedNodeByRootMap

### PR DESCRIPTION
**Pull request by int88**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

delete from s.nodeByPayload as well when pruneFinalizedNodeByRootMap

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

**Comment by int88** [comment]
@potuz @prestonvanloon PTAL :)

**Review by potuz — COMMENTED** [review]
Nice catch, a regression test would be good though

**Comment by int88** [comment]
@potuz updated.

File: beacon-chain/forkchoice/doubly-linked-tree/store_test.go [hunk 1]
```diff
@@ -235,6 +235,7 @@ func TestStore_Prune_NoDanglingBranch(t *testing.T) {
 	s.finalizedCheckpoint.Root = indexToHash(1)
 	require.NoError(t, s.prune(context.Background()))
 	require.Equal(t, len(s.nodeByRoot), 1)
+	require.Equal(t, len(s.nodeByPayload), 1)
```
Comment (potuz) [inline beacon-chain/forkchoice/doubly-linked-tree/store_test.go:238]: You need to use different payload hashes for this test to make sense

Comment (int88) [reply beacon-chain/forkchoice/doubly-linked-tree/store_test.go:238]: @potuz  updated.

**Review by rauljordan — APPROVED** [review]
LGTM! :thumbsup:

**Pull request merged**